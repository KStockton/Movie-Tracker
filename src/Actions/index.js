export const addTopMovies = topMovies => ({
    type: "ADD_TOP_MOVIES",
    topMovies
})