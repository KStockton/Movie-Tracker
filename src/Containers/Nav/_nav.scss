@import "../../Sass/variables";

nav {
  background-color: $nav-background-color;
  height: 6em;
  @include displayFlex(center, center, null, row);
  border-bottom: 0.2em solid $user-form-border-color;
}
ul {
  list-style: none;
  width: 95%;
  @include displayFlex(space-between, center, center, row);
}
nav li {
  height: 3em;
  @include displayFlex(center, center, center, row);
  border-radius: 1em;
  padding: 0 1em;
}

a.nav-link,
button.nav-link {
  text-decoration: none;
  color: $text-color-main;
  font-size: $nav-button-font-size;
  font-weight: bold;
  font-size: 1em;
  background-color: transparent;
  border: 0.1em $user-form-border-color solid;
  padding: 0.5em;
  transition: color 0.2s;
}
a.nav-link:hover,
button.nav-link:hover {
  color: $nav-text-color-hover;
}
.main-nav-container {
  width: 40%;
  @include displayFlex(flex-start, center, baseline, row);
}
.nav-title {
  margin: 0 0.8em 0 0;
  color: $text-color-main;
}

.nav-signed-in-user {
  color: $text-color-main;
  padding: 1em;
}
.nav-user-name {
  color: $nav-text-color-hover;
}
.nav-signed-in-container {
}
.nav-title.signout-button {
  background-color: transparent;
  border: 0.1em $user-form-border-color solid;
  padding: 0.5em;
  font-size: 1em;
  font-weight: bold;
}

.nav-title.signout-button:hover {
  cursor: pointer;
  color: $nav-text-color-hover;
}
