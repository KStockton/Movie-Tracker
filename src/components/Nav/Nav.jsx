import React, { Component } from "react";
import { BrowserRouter as Router, Route, Link } from "react-router-dom";
import { connect } from "react-redux";

class Nav extends Component {
  constructor(props) {
    super(props);

    this.state = {};
  }

  render() {
    return this.props.user.name === undefined ? (
      <nav>
        <ul>
          <li>
            <Link to="/home">Home</Link>
          </li>
          <section className="sign-in-container" />
          <li>
            <Link to="/login">Log in or Sign up</Link>
          </li>
        </ul>
      </nav>
    ) : (
      <nav>
        <ul>
          <li>
            <Link to="/home">Home</Link>
          </li>
          <section className="sign-in-container" />
          <li>
            <Link to="/user-settings">User</Link>
          </li>
        </ul>
      </nav>
    );
  }
}

const mapStateToProps = state => ({
  user: state.user
});

export default connect(mapStateToProps)(Nav);
