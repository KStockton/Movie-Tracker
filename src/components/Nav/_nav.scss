nav {
    background-color: darkgrey;
    height: 2.5em;
    display: flex;
    justify-content: space-between;
}
ul {
    display:flex;
    list-style: none;
}