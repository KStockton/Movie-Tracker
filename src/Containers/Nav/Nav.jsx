import React, { Component } from "react";
import { NavLink } from "react-router-dom";
import { connect } from "react-redux";
import { removeUser } from "../../Actions";
import PropTypes from 'prop-types';


export class Nav extends Component {
  render() {
    return this.props.user.name === undefined ? (
      <nav>
        <ul>
          <div className="main-nav-container">
            <h1 className="nav-title">Movie Tracker</h1>
            <li>
              <NavLink to="/" className="nav-link">
                Home
              </NavLink>
            </li>
          </div>
          <section className="sign-in-container" />
          <li>
            <NavLink to="/login" className="nav-link">
              Log in/Sign up
            </NavLink>
          </li>
        </ul>
      </nav>
    ) : (
      <nav>
        <ul>
          <div className="main-nav-container">
            <h1 className="nav-title">Movie Tracker</h1>
            <li>
              <NavLink to="/" className="nav-link">
                Home
              </NavLink>
            </li>
            <li>
              <NavLink to="/favorites" className="nav-link">
                Favorites
              </NavLink>
            </li>
          </div>
          <section className="sign-in-container" />

          <li className="nav-signed-in-container">
            <span className="nav-signed-in-user">
              Signed in as:{" "}
              <span className="nav-user-name">{this.props.user.name} </span>
            </span>
            <button
              className="nav-title signout-button"
              onClick={() => this.props.removeUser({})}
            >
              Sign Out
            </button>
          </li>
        </ul>
      </nav>
    );
  }
}

export const mapStateToProps = state => ({
  user: state.user
});
export const mapDispatchToProps = dispatch => ({
  removeUser: user => dispatch(removeUser(user))
});

Nav.propTypes = {
  user: PropTypes.object,
  removeUser: PropTypes.func
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Nav);
