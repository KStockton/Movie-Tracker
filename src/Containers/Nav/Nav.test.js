import React from "react";
import { Nav, mapStateToProps, mapDispatchToProps } from "./Nav";
import {removeUser} from '../../Actions/'
import { shallow } from "enzyme";

describe("Nav", () => {
  let wrapper;
  const mockProps = { user: { name: "max", id: 1 }, removeUser: jest.fn() };
  beforeEach(() => {
    wrapper = shallow(<Nav {...mockProps} />);
  });

  it("Nav component should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("should return a props object ", () => {
    const mockState = { user: { name: "max", id: 1 }, removeUser: jest.fn() };
    const expected = {
      user: mockState.user,
    };
    const mappedProps = mapStateToProps(mockState);
    expect(mappedProps).toEqual(expected);
  });
  it('should call dispatch when using a function from MDTP', () => {
    const mockDispatch = jest.fn()
    const actionToDispatch = removeUser({})
    const mappedProps = mapDispatchToProps(mockDispatch)
    mappedProps.removeUser({})
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
  })
});
