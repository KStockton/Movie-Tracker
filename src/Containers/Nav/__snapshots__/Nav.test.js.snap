// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Nav component should match the snapshot 1`] = `
<nav>
  <ul>
    <div
      className="main-nav-container"
    >
      <h1
        className="nav-title"
      >
        Movie Tracker
      </h1>
      <li>
        <NavLink
          className="nav-link"
          to="/"
        >
          Home
        </NavLink>
      </li>
      <li>
        <NavLink
          className="nav-link"
          to="/favorites"
        >
          Favorites
        </NavLink>
      </li>
    </div>
    <section
      className="sign-in-container"
    />
    <li
      className="nav-signed-in-container"
    >
      <span
        className="nav-signed-in-user"
      >
        Signed in as:
         
        <span
          className="nav-user-name"
        >
          max
           
        </span>
      </span>
      <button
        className="nav-title signout-button"
        onClick={[Function]}
      >
        Sign Out
      </button>
    </li>
  </ul>
</nav>
`;
